src/random_ray/flat_source_domain.cpp

```diff
@@ -351,6 +353,27 @@ void FlatSourceDomain::compute_k_eff()
 
   double k_eff_new = k_eff_ * (fission_rate_new / fission_rate_old);
 
+  // Normalize fluxes by total number of fission neutrons produced. This ensures
+  // consistent scaling of the eigenvector such that its magnitude is
+  // comparable to the eigenvector produced by the Monte Carlo solver.
+  // Multiplying by the eigenvalue is unintuitive, but it is necessary.
+  // If the eigenvalue is 1.2, per starting source neutron, you will
+  // generate 1.2 neutrons. Thus if we normalize to generating only ONE neutron
+  // in total for the whole domain, then we don't actually have enough flux to
+  // generate the required 1.2 neutrons. We only know the flux required to
+  // generate 1 neutron (which would have required less than one starting
+  // neutron). Thus, you have to scale the flux up by the eigenvalue such
+  // that 1.2 neutrons are generated, so as to be consistent with the
+  // bookkeeping in MC which is all done per starting source neutron (not per
+  // neutron produced).
+  double total_fission_neutrons = fission_rate_new * simulation_volume_;
+  double norm_factor = k_eff_new / total_fission_neutrons;
```

Comment:
Does this make sense?

norm_factor = k_eff_new / total_fission_neutrons
                       = (k_eff_ * (fission_rate_new / fission_rate_old)) / (fission_rate_new * simulation_volume_)
                       = k_eff_ / (fission_rate_old * simulation_volume_)